source/node/archiveManagement/marshalling.js

```diff
@@ -47,26 +44,29 @@ function credentialsToSource(sourceCredentials, archiveCredentials, initialise =
             const datasource = result.datasource;
             const defaultArchive = Archive.createWithDefaults();
             return initialise
-                ? datasource.save(defaultArchive, archiveCredentials).then(() =>
+                ? datasource.save(defaultArchive.getHistory(), archiveCredentials).then(() =>
                       Object.assign(
                           {
                               archive: defaultArchive
                           },
                           result
                       )
                   )
-                : datasource.load(archiveCredentials).then(archive =>
-                      Object.assign(
-                          {
-                              archive
-                          },
-                          result
-                      )
-                  );
+                : datasource
+                      .load(archiveCredentials)
+                      .then(history => Archive.createFromHistory(history))
+                      .then(archive =>
+                          Object.assign(
```

Comment:
Do we have object spread enabled in the babel so we can use that instead of `Object.assign`?

Reply:
Nope.. Core isn't transpiled on publishing currently. Prettier is going to town on this function though, haha.. it's quite ugly. Perhaps we can add babel to the process at some later point :)